src/Microsoft.PowerShell.Commands.Utility/commands/utility/GetHash.cs

```diff
@@ -140,12 +140,22 @@ protected override void ProcessRecord()
                 }
                 catch (FileNotFoundException ex)
                 {
-                    ErrorRecord errorRecord = new ErrorRecord(ex,
+                    var errorRecord = new ErrorRecord(
+                        ex,
                         "FileNotFound",
                         ErrorCategory.ObjectNotFound,
                         path);
                     WriteError(errorRecord);
                 }
+                catch (UnauthorizedAccessException ex)
+                {
+                    var errorRecord = new ErrorRecord(
+                        ex,
+                        "UnauthorizedAccessError",
+                        ErrorCategory.InvalidData,
+                        path);
+                    WriteError(errorRecord);
+                }
```

Comment:
Do you think it's worth doing a general catch or additional catches here as well? Looking at the exceptions that can be thrown from [`System.IO.File.OpenRead`](https://docs.microsoft.com/en-us/dotnet/api/system.io.file.openread?view=netcore-3.0#exceptions) it seems likely we can run into the following exceptions somewhat commonly as well:

- `PathTooLongException`
- `IOException`

Reply:
IOException combines  FileNotFoundException, DirectoryNotFoundException, and PathTooLongException.

I guess PathTooLongException will be raised above at path resolving time.

So nothing to do. :-)